dask/dataframe/core.py

```diff
@@ -3316,15 +3316,17 @@ def cummin(self, axis=None, skipna=True, out=None):
         )
 
     def _validate_condition(self, cond):
+        cond_res = cond(self._meta) if callable(cond) else cond
         if not (
-            is_dask_collection(cond)
-            or is_dataframe_like(cond)
-            or is_series_like(cond)
-            or is_index_like(cond)
+            is_dask_collection(cond_res)
+            or is_dataframe_like(cond_res)
+            or is_series_like(cond_res)
+            or is_index_like(cond_res)
+            or callable(cond_res)
```

Comment:
Thanks for taking my review into account! I think this line should probably be removed then ? (the callable result shouldn't be a callable itself)

Reply:
👍